### ISSUE TITLE
How to mock the provider response instead of using the real provider 

### ISSUE THREAD

**Issue by miguel-wr** [pact-foundation/pact-net#269]
I need to mock the provider response instead of using the real provider like in Pact JVM with junit
https://docs.pact.io/implementation_guides/jvm/provider/junit

**Comment by mefellows** [comment]
Hi @miguel-wr, I'm sorry I'm not sure what you're asking. 

On the consumer side, you write a pact test that mocks the provider.

On the provider side, you have to talk to the real provider (the link you gave is a provider test for JVM, which absolutely must communicate to a real provider).

You may want to consider running the .NET workshop here: https://docs.pact.io/implementation_guides/workshops

**Comment by miguel-wr** [comment]
Thanks a lot for your quick answer. 
For example, for JVM, we can use a MockBean and mock the response like this
    @MockBean
    private RecipientRepository recipientRepository;

    @TestTemplate
    void testTemplate(PactVerificationContext context) {
        context.verifyInteraction();
    }

    @State("get a valid phone number")
    public void getRecipientsByPhoneReturnsRecipientsList() {
        given(someRepository.findByPhone(anyString())).willReturn(buildList());
    
Is it possible to do the same thing with pact-net? 
Thanks a lot

**Comment by mefellows** [comment]
Right, so that sounds like a general .NET question to me rather than something for Pact. Am I missing something?

Unfortunately, I know very little about .NET so won't be able to give you any good advice here outside of Pact specific things.
